db/migrate/20150525081931_remove_duplicate_tokens.rb

```diff
@@ -0,0 +1,28 @@
+class RemoveDuplicateTokens < ActiveRecord::Migration
+  def up
+    # only managed hosts should have tokens
+    Token.where(:id => Token.joins("left outer join hosts on hosts.id = tokens.host_id").
+                             where("hosts.type != 'Host::Managed'").pluck('tokens.id')).delete_all
+
+    # deleting duplicate tokens
+    hosts_with_duplicate_tokens = Token.having('count(*) > 1').group(:host_id).select('host_id')
+    existing_tokens = hosts_with_duplicate_tokens.map(&:token).compact
```

Comment:
Yeah, I don't know which is going to be more correct.. it's probably going to depend a lot on how consistent the SQL query responses are, which varies depending on the server.  It might be they're both inaccurate if the SQL server changes the ordering.